internal/proxy/origins/prometheus/matrix.go

```diff
@@ -318,8 +382,14 @@ func (me *MatrixEnvelope) SeriesCount() int {
 // ValueCount returns the count of all values across all Series in the Timeseries object
 func (me *MatrixEnvelope) ValueCount() int {
 	c := 0
+	wg := sync.WaitGroup{}
 	for i := range me.Data.Result {
-		c += len(me.Data.Result[i].Values)
+		wg.Add(1)
+		go func(j int) {
+			c += j
```

Comment:
Each of the goroutines per me.Data.Result can be updating t concurrently because the lock is specific per me.Data.Result; you are not guaranteed to get the correct value where you want it.

Reply:
is this comment attached to the correct line number?

Reply:
(Looks like you intended this for L170, i got it covered)

Reply:
Yeah, sorry. This relates to `c`. There is concurrent access of `c` with no thread safety.

Reply:
👍